File: FantasyRadio/src/main/java/ru/sigil/fantasyradio/SectionsPagerAdapter.java
package ru.sigil.fantasyradio;

/**
 * Created by namelessone
 * on 13.06.17.
 */

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import ru.sigil.fantasyradio.archieve.ArchieveFragment;
import ru.sigil.fantasyradio.saved.SavedFragment;
import ru.sigil.fantasyradio.schedule.ScheduleFragment;

/**
 * A {@link FragmentPagerAdapter} that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class SectionsPagerAdapter extends FragmentPagerAdapter {
    private RadioFragment radioFragment = new RadioFragment();
    private ScheduleFragment scheduleFragment = new ScheduleFragment();
    private ArchieveFragment archieveFragment = new ArchieveFragment();
    private SavedFragment savedFragment = new SavedFragment();
    private Context context;

    public SectionsPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        // getItem is called to instantiate the fragment for the given page.
        // Return a DummySectionFragment (defined as a static inner class
        // below) with the page number as its lone argument.
        //TODO
        Fragment fragment = new Fragment();
        switch (position) {
            case 0:
                fragment = radioFragment;
                break;
            case 1:
                fragment = scheduleFragment;
                break;
            case 2:
                fragment = archieveFragment;
                break;
            case 3:
                fragment = savedFragment;
                break;
        }
        return fragment;
    }

    @Override
    public int getCount() {
        return 4;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return context.getString(R.string.first_tab_text);
            case 1:
                return context.getString(R.string.second_tab_text);
            case 2:
                return context.getString(R.string.third_tab_text);
            case 3:
                return context.getString(R.string.fourth_tab_text);
        }
        return null;
    }

    public void notifySavedFragment() {
        savedFragment.notifyAdapter();
    }
}

File: FantasyRadio/src/main/kotlin/ru/sigil/fantasyradio/SectionsPagerAdapter.kt
package ru.sigil.fantasyradio

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import ru.sigil.fantasyradio.saved.SavedFragment
import ru.sigil.fantasyradio.archive.ArchiveFragment
import ru.sigil.fantasyradio.schedule.ScheduleFragment



/**
 * A {@link FragmentPagerAdapter} that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
class SectionsPagerAdapter(private val context: Context, fragmentManager: FragmentManager): FragmentPagerAdapter(fragmentManager) {
    private val radioFragment = RadioFragment()
    private val scheduleFragment = ScheduleFragment()
    private val archiveFragment = ArchiveFragment()
    private val savedFragment = SavedFragment()

    override fun getItem(position: Int): Fragment {
        // getItem is called to instantiate the fragment for the given page.
        // Return a DummySectionFragment (defined as a static inner class
        // below) with the page number as its lone argument.
        //TODO
        var fragment = Fragment()
        when (position) {
            0 -> fragment = radioFragment
            1 -> fragment = scheduleFragment
            2 -> fragment = archiveFragment
            3 -> fragment = savedFragment
        }
        return fragment
    }

    override fun getCount(): Int {
        return 4
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return context.getString(R.string.first_tab_text)
            1 -> return context.getString(R.string.second_tab_text)
            2 -> return context.getString(R.string.third_tab_text)
            3 -> return context.getString(R.string.fourth_tab_text)
        }
        return null
    }

    fun notifySavedFragment() {
        savedFragment.notifyAdapter()
    }
}